hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +78,973 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+                        = __scalar_as_int(h_postype.data[central_idx].w);
+                    rigid_types.insert(central_type);
+                    }
+                }
+            }
+
+        // go through each molecule and check if the center of mass is in right location
+        if (m_rigid_bodies && !rigid_types.empty())
+            {
+            const Scalar tol(1e-6);
+            // access molecule order
+            ArrayHandle<unsigned int> h_molecule_order(m_rigid_bodies->getMoleculeOrder(),
+                                                       access_location::host,
+                                                       access_mode::read);
+            ArrayHandle<unsigned int> h_molecule_len(m_rigid_bodies->getMoleculeLengths(),
+                                                     access_location::host,
+                                                     access_mode::read);
+            ArrayHandle<unsigned int> h_molecule_idx(m_rigid_bodies->getMoleculeIndex(),
+                                                     access_location::host,
+                                                     access_mode::read);
+            ArrayHandle<unsigned int> h_molecule_list(m_rigid_bodies->getMoleculeList(),
+                                                      access_location::host,
+                                                      access_mode::read);
+            Index2D molecule_indexer = m_rigid_bodies->getMoleculeIndexer();
+            unsigned int nmol = molecule_indexer.getH();
+            // access body definitions
+            ArrayHandle<Scalar3> h_body_pos(m_rigid_bodies->getBodyOffsets(),
+                                            access_location::host,
+                                            access_mode::read);
+            Index2D h_body_idx = m_rigid_bodies->getBodyIndexer();
+            // accesss particle data
+            ArrayHandle<Scalar4> h_postype(m_pdata->getPositions(),
+                                           access_location::host,
+                                           access_mode::read);
+            ArrayHandle<Scalar4> h_velocity(m_pdata->getVelocities(),
+                                            access_location::host,
+                                            access_mode::read);
+            ArrayHandle<unsigned int> h_body(m_pdata->getBodies(),
+                                             access_location::host,
+                                             access_mode::read);
+            ArrayHandle<unsigned int> h_rtag(m_pdata->getRTags(),
+                                             access_location::host,
+                                             access_mode::read);
+            // loop over all molecules
+            unsigned int n_particles_local = m_pdata->getN() + m_pdata->getNGhosts();
+            for (unsigned int ibody = 0;
+                 ibody < nmol && !(invalid_center_of_mass || rigid_types.empty());
```

Comment:
Why the check on `rigid_types.empty()` here? It seems like that part of the conditions will always end up being true because it is in the if above, then the set isn't modified in the loop.